synapse/types.py

```diff
@@ -240,7 +242,8 @@ def strip_invalid_mxid_characters(localpart):
     Returns:
         localpart (basestring): the localpart having been stripped
     """
-    return filter(lambda c: c in mxid_localpart_allowed_characters, localpart)
+    filtered = filter(lambda c: c in mxid_localpart_allowed_characters, localpart)
+    return "".join(list(filtered))
```

Comment:
I don't think the cast to `list` is required?

Reply:
Oh, indeed.